File: packages/shared/ports.ts
/**
 * SINGLE SOURCE OF TRUTH FOR ALL PORTS
 * 
 * MUST MATCH: oneclaw-node/src/ports.rs (Rust is authoritative)
 * 
 * DO NOT use environment variables for ports.
 * DO NOT use CLI arguments for ports (except daemon --port for testing).
 * DO NOT use config files for ports.
 * 
 * If you need to change a port, change it in BOTH places:
 *   1. oneclaw-node/src/ports.rs (Rust)
 *   2. packages/shared/ports.ts (TypeScript)
 */

export const PORTS = {
  /** Harness - workflow execution engine */
  HARNESS: 9000,  // ✅ Matches ports.rs HARNESS_PORT
  
  /** Daemon - Rust gateway/proxy */
  DAEMON: 8787,   // ✅ Matches ports.rs DAEMON_PORT
  
  /** API - REST API server */
  API: 3000,      // ✅ Matches ports.rs API_PORT
} as const;

export const INTERNAL_URLS = {
  HARNESS: `http://localhost:${PORTS.HARNESS}`,
  DAEMON: `http://localhost:${PORTS.DAEMON}`,
  API: `http://localhost:${PORTS.API}`,
} as const;

File: oneclaw-node/src/ports.rs
/// PORT CONFIGURATION
/// 
/// Daemon port is configurable via CLI: `daemon --port 8787`
/// Harness and API ports are hardcoded below.
/// 
/// If you need to change Harness/API ports, change them HERE and rebuild.

/// Daemon - Rust agent runtime (Web UI)
/// Default: 8787, but configurable via CLI
pub const DAEMON_PORT: u16 = 8787;

/// Harness - workflow execution engine (Tools API)
/// Hardcoded - daemon expects this
pub const HARNESS_PORT: u16 = 9000;

/// API - REST API server (OAuth proxy)
/// Hardcoded
pub const API_PORT: u16 = 3000;

/// Internal URLs (used by daemon to call harness/api)
pub const DAEMON_URL: &str = "http://localhost:8787";
pub const HARNESS_URL: &str = "http://localhost:9000";
pub const API_URL: &str = "http://localhost:3000";
